flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/ExecutionFailureHandler.java

```diff
@@ -62,18 +71,41 @@ public ExecutionFailureHandler(
 	 * @return result of the failure handling
 	 */
 	public FailureHandlingResult getFailureHandlingResult(ExecutionVertexID failedTask, Throwable cause) {
+		return handleFailure(cause, () -> failoverStrategy.getTasksNeedingRestart(failedTask, cause));
+	}
+
+	/**
+	 * Return result of failure handling on a global failure. Can be a set of task vertices to restart
+	 * and a delay of the restarting. Or that the failure is not recoverable and the reason for it.
+	 *
+	 * @param cause of the task failure
+	 * @return result of the failure handling
+	 */
+	public FailureHandlingResult getGlobalFailureHandlingResult(final Throwable cause) {
+		return handleFailure(
+			cause,
+			() -> StreamSupport
+				.stream(failoverTopology.getFailoverVertices().spliterator(), false)
+				.map(FailoverVertex::getExecutionVertexID)
+				.collect(Collectors.toSet()));
+	}
+
+	private FailureHandlingResult handleFailure(
+			final Throwable cause,
+			final Supplier<Set<ExecutionVertexID>> vertexToRestartSupplier) {
```

Comment:
I don't think the `Supplier` is needed. We can pass the Set of vertices instead.

Reply:
You are right. Done.